### PR TITLE
Add Ecole project

### PR DESCRIPTION
**Pull request by AntoinePrv**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] ~If static libraries are linked in, the license of the static library is packaged.~
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

----------

 - The package is automatically vendoring [span-lite](https://github.com/martinmoene/span-lite) and [range-v3](https://github.com/ericniebler/range-v3). Should I turn them into packages, or simply package their license?
 - Should I package license for conda header only libraries, _e.g._ xtensor?

### CODE REVIEW

**Comment by gasse** [comment]
Hi, I agree to be a maintainer for this recipe, along with @AntoinePrv

**Comment by AntoinePrv** [comment]
MacOS build is failing because we use of `clock_gettime(CLOCK_PROCESS_CPUTIME_ID, ...)` which comes in `MacOS >=10.12` I believe.

@conda-forge/core the [documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html#requiring-newer-macos-sdks) suggest contacting you before increasing `MACOSX_SDK_VERSION` and `MACOSX_DEPLOYMENT_TARGET`.
Is this what we should be using? Our library is `C++17`, so if not for `clock_gettime`, I believe we would have to increasing it one way or the other.

**Comment by isuruf** [comment]
Yes, that's fine. You only have to update `MACOSX_DEPLOYMENT_TARGET` to 10.12

**Comment by AntoinePrv** [comment]
Increasing again `MACOSX_SDK_VERSION` to 10.13 for `std::variant` 
```
std::variant<...> is unavailable: introduced in macOS 10.13
```

**Comment by isuruf** [comment]
For that you can add `-D_LIBCPP_DISABLE_AVAILABILITY` to CXXFLAGS

File: recipes/ecole/build.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o pipefail
+set -o nounset
+
+
+# For using C++17 symbols on Apple before they are officially released
+if [[ "$OSTYPE" == "darwin"* ]]; then
```
Comment (isuruf) [inline recipes/ecole/build.sh:9]: use `$target_platform == osx-*`

**Comment by AntoinePrv** [comment]
@isuruf Thanks for taking the time to look at this PR. How can we move it forward?

**Pull request merged**